servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpDataSourceTransformations.java

```diff
@@ -433,4 +437,12 @@ static RuntimeException newTrailersExpectedButNotSeenException() {
             }
         });
     }
+
+    private static final class AggregationException extends IllegalStateException {
```

Comment:
I question the value of this change after https://github.com/netty/netty/pull/10330 is available. The only benefit I see is that the name of the exception is different in the logs. As the class is private, users can not use the exception to decide further actions. Is that what your intent is? 

If we create a publicly accessible exception class like `TooLongHttpPayloadException` then the change makes more sense to me.

Reply:
> The confusion was that IllegalArgumentException is thrown when users don't pass any argument. A more clear instance of IllegalStateException makes more sense in this case.

The benefit is trivial IMO.

> I'm ok with making the exception type public.

My suggestion is to narrow the scope of the exception (only when there is a buffer overflow) and then make it public